### PR TITLE
Source Hubspot: PropertyHistopry stream to support incremental

### LINKED ISSUE #32694: Extend PropertyHistory stream to support Incremental sync
## Relevant information
The `PropertyHistory` stream does not support incremental because an [endpoint](https://legacydocs.hubspot.com/docs/methods/contacts/get_contacts) it uses does not support filtering by date. The only way to support the feature is to use semi-incremental technique which means we need to filter out records that not satisfy checkpont condition.

## Implementation plan
- Investigate whether incremental feature available for given stream
- Implement the feature if possible

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
*`PropertyHistory` stream to be extended to support Incremental https://github.com/airbytehq/airbyte/issues/32694*

## How
- *Semi-incremental implemented to filter out records not satisfied checkpoint condition*
- *Sample files updated: state to new format*
- *Expected records updated*

### CODE REVIEW

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 1]
```diff
@@ -1783,6 +1783,10 @@ class PropertyHistory(Stream):
     scopes = {"crm.objects.contacts.read"}
     properties_scopes = {"crm.schemas.contacts.read"}
 
+    @property
+    def cursor_field_datetime_format(self):
+        return "x"
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:1788]: can you give an example of the cursor value?

Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:1788]: Cursor value is a timestamp in milliseconds (`1700681340514`) and is supported in this function: [`filter_by_state`](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L837)

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:1788]: Can you add the part of comment about timestamp from doc string for this method also.

**Review by lazebnyi — APPROVED** [review]
Need add small doc string, but other stuff looking good.

File: docs/integrations/sources/hubspot.md [hunk 2]
```diff
@@ -221,6 +221,7 @@ becomes
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                                                            |
 |:--------|:-----------|:---------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1.8.0   | 2023-11-23 | [32778](https://github.com/airbytehq/airbyte/pull/32778) | Extend `PropertyHistory` stream to support Incremental                                                                                                                             |
```
Comment (katmarkham) [inline docs/integrations/sources/hubspot.md:224]: Extend `PropertyHistory` stream to support incremental sync

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py (hunk 1 above)
Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:1789]: Done

**Review by lazebnyi — APPROVED** [review]
lgtm

**Pull request merged**